core/runtime/src/main/java/io/quarkus/runtime/ApplicationLifecycleManager.java

```diff
@@ -142,12 +142,8 @@ public static final void run(Application application, Class<? extends QuarkusApp
             } finally {
                 stateLock.unlock();
             }
-            application.stop(new Runnable() {
```

Comment:
I think this will break some tests, although I can't remember the details. Maybe the default handler that calls System.exit() needs to do it from a new thread.

Reply:
Yeah I thought so as well, but when I ran the CI on my fork, all tests passed.

So should I leave it as is (assuming we'll get green CI here as well) or change to use another thread?

Reply:
I know I had a good reason for doing it this way, but I can't actually remember what it was. Maybe subsequent changes mean it is not needed now, so if CI passes then lets merge, and if we see intermittent failures we can change it.

Reply:
Cool, makes sense